src/semgrep_agent.py

```diff
@@ -55,7 +55,10 @@ def __init__(self, option_strings, dest, **kwargs) -> None:  # type: ignore
     def __call__(self, _, namespace, values, option_string=None) -> None:  # type: ignore
         envvar = FLAG_TO_ENV.get(option_string)
         if envvar:
-            os.environ[envvar] = values
+            if envvar in os.environ:
+                os.environ[envvar] += " " + values
```

Comment:
Sorry about the oversight of missing this, and thank you for fixing it! I worry though that we might have some users who have set both a flag and an env var for the same option, and this would break their CI jobs now 🤔 

Specifically, imagine if someone committed this (silly) configuration: `SEMGREP_APP_TOKEN=123 semgrep ci --publish-token=123`. It'd previously work correctly, and would break after this as token would be set to `"123 123"`.

I think I'm comfortable with setting a condition where we have this behavior only if the envvar is `SEMGREP_RULES`. Care to update the PR to do that instead? Otherwise, I could just fix up your PR and merge.

Reply:
I have added a commit that restricts the multiple values to `--config` and `--audit-on`. I included `--audit-on` since that also has `multiple=True` in [semgrep ci.py](https://github.com/returntocorp/semgrep/blob/develop/semgrep/semgrep/commands/ci.py#L136).

Reply:
Thanks so much :heart: